components/collections-panel/collections-panel-component.js

```diff
@@ -0,0 +1,147 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { toastr } from 'react-redux-toastr';
+
+import CollectionPanelItem from './collections-panel-item/collections-panel-item-component';
+
+// constants
+import { FAVOURITES_COLLECTION } from './collections-panel-constants';
+
+// styles
+import styles from './collections-panel-styles.scss';
+
+class CollectionsPanel extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      newCollection: null // new collection's name
+    };
+  }
+
+  onAddCollection() {
+    const { addCollection } = this.props;
+    const { newCollection } = this.state;
+
+    if ((newCollection || '').toLowerCase() === 'favourites') {
+      toastr.error('Duplicated Favourites list', 'You cannot duplicate this list.');
+      return;
+    }
+
+    addCollection(newCollection);
+  }
+
+  onToggleFavourite() {
+    const { toggleFavourite, favourites, resource, resourceType } = this.props;
+    const favourite = favourites.find(fav => fav.resourceId === resource.id) || {};
+    toggleFavourite(favourite, { resourceId: resource.id, resourceType });
+  }
+
+  onToggleResource(isAdded, collection) {
+    const { toggleResource, resource, resourceType } = this.props;
+    toggleResource(isAdded, collection.id, { id: resource.id, type: resourceType });
+  }
+
+  hanldeKeyPress(key) {
+    if (key !== 'Enter') return;
+
+    this.onAddCollection();
+  }
+
+  renderCollections() {
+    const { collections, resource, resourceType, favourites } = this.props;
+
+    const favouriteCollection = (
+      <CollectionPanelItem
+        key={FAVOURITES_COLLECTION.id}
+        collection={FAVOURITES_COLLECTION}
+        resource={resource}
+        resourceType={resourceType}
+        isChecked={favourites.some(favourite =>
+          favourite.resourceId === resource.id)}
+        onToggleResource={() => this.onToggleFavourite()}
```

Comment:
You shouldn't use arrow functions as props of the component. Instead you should 
```
onToggle={this.onToggleFavourite}
```
and then in the function declaration
```
onToggleFavourite = () => {}
```